plasmapy/formulary/collisions.py

```diff
@@ -2196,52 +2248,74 @@ def __init__(
         # Note: This function uses CGS units internally to coincide with our references.
         # Input is taken in MKS units and then converted as necessary. Output is in MKS units.
 
-        if v_a is None:
-            if T_a is not None:
-                v_a = thermal_speed(T_a, test_particle)
-            else:
-                raise ValueError("Please specify either v_a or T_a.")
-        elif T_a is not None:
-            raise ValueError("Please specify either v_a or T_a, not both.")
-
         if (
-            isinstance(v_a, np.ndarray)
+            isinstance(v_drift, np.ndarray)
             and isinstance(n_b, np.ndarray)
-            and v_a.shape != n_b.shape
+            and v_drift.shape != n_b.shape
         ):
             raise ValueError("Please specify arrays of equal length.")
 
         self.test_particle = test_particle
         self.field_particle = field_particle
-        self.v_a = v_a
-        self.T_a = (
-            T_a
-            if T_a is not None
-            else (0.5 * test_particle.mass * v_a**2 / k_B).to(u.K)
-        )
-        self.n_b = n_b
+        self.v_drift = v_drift
+        self.T_a = T_a
+        self.n_a = n_a
         self.T_b = T_b
+        self.n_b = n_b
         self.Coulomb_log = (
             Coulomb_log
             if isinstance(Coulomb_log, u.Quantity)
             else Coulomb_log * u.dimensionless_unscaled
         )
 
+    @cached_property
+    @validate_class_attributes(expected_attributes=["T_a"])
+    def _v_T_a(self):
+        """
+        The thermal velocity of the test particle species.
+        """
+
+        return thermal_speed(self.T_a, self.test_particle)
+
+    @cached_property
+    @validate_class_attributes(both_or_either_attributes=[("T_a", "v_drift")])
+    def _v_a(self):
+        """
+        The net velocity of the test particle species due to both drift and thermal motions.
+        If T_a is not specified, then the net velocity is assumed to be dominated by drift.
+        If v_drift is not specified, then the net velocity is assumed to be dominated by thermal motion.
+        """
+
+        if self.T_a is not None and self.v_drift is not None:
+            return self._v_T_a + self.v_drift
```

Comment:
This should be V_T_ss' as opposed to just a straight-up addition, right?

Reply:
In the Maxwellian ones it looks like yes, but in the original ones it should be either v_drift or v_T_b (or maybe we should be taking the mean of those too?))